File: U1/EXAMEN_SOAP_DOTNET_GR01/01 SERVIDOR/BanquitoServer_Soap_DotNet_GR01/BanquitoServer_Soap_DotNet_GR01/Models/Movimiento.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BanquitoServer_Soap_DotNet_GR01.Models
{
    /// <summary>
    /// Entidad Movimiento (depósitos y retiros)
    /// </summary>
    [Table("Movimiento")]
    public class Movimiento
    {
        [Key]
        [Column("MovimientoId")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long MovimientoId { get; set; }

        [Required]
        [Column("TipoMovimiento")]
        [MaxLength(20)]
        [Index("IX_Movimiento_Tipo")]
        public string TipoMovimiento { get; set; }

        [Required]
        [Column("Monto")]
        public decimal Monto { get; set; }

        [Required]
        [Column("FechaMovimiento")]
        [Index("IX_Movimiento_Fecha")]
        public DateTime FechaMovimiento { get; set; }

        [Column("Descripcion")]
        [MaxLength(200)]
        public string Descripcion { get; set; }

        [Column("SaldoAnterior")]
        public decimal? SaldoAnterior { get; set; }

        [Column("SaldoNuevo")]
        public decimal? SaldoNuevo { get; set; }

        // Foreign Key
        [Required]
        [Column("CuentaId")]
        [ForeignKey("Cuenta")]
        [Index("IX_Movimiento_CuentaId")]
        public long CuentaId { get; set; }

        // Propiedad de navegación
        public virtual Cuenta Cuenta { get; set; }
    }
}

File: U2/PROYECTO/BanquitoServer_Restfull_Java_GR01/src/main/java/ec/edu/monster/Models/Movimiento.java
package ec.edu.monster.Models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * Entidad Movimiento bancario
 */
@Entity
@Table(name = "Movimiento")
@Data
@NoArgsConstructor
public class Movimiento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "MovimientoId")
    private Long movimientoId;

    @Column(name = "TipoMovimiento", nullable = false, length = 20)
    private String tipoMovimiento;

    @Column(name = "Monto", nullable = false, precision = 12, scale = 2)
    private BigDecimal monto;

    @Column(name = "FechaMovimiento", nullable = false)
    private LocalDateTime fechaMovimiento;

    @Column(name = "SaldoAnterior", precision = 12, scale = 2)
    private BigDecimal saldoAnterior;

    @Column(name = "SaldoNuevo", precision = 12, scale = 2)
    private BigDecimal saldoNuevo;

    @Column(name = "Descripcion", length = 200)
    private String descripcion;

    // Foreign Key
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CuentaId", nullable = false)
    private Cuenta cuenta;
}
